### PR TITLE
Add a function `htmlOrXmlCoarseParse` to extract content from html or xml format string.

### PR DESCRIPTION
**Pull request by zlx19950903**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category (leave one):

- New Feature

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
- Mentioned in #18454
- add function `htmlOrxmlCoarseParse`;
- support `<script></script>` parse;
- support `<style></style>` parse;
- support `<![CDATA[]]>` parse;
- support white space collapse;
- support any `<content>` format parse;
- HyperScan to support SIMD;
- Everything is done in a single pass.

Usage:
```
SELECT htmlOrXmlCoarseParse('<script>hello </script><![CDATA[world]]>')
world
```

### CODE REVIEW

**Comment by zlx19950903** [comment]
I got a question about fast test.  My source code depends on a third party named hyperscan. When I build locally, USE_HYPERSCAN will be set default 1. But in the fast test, it is set to be 0. Why does it occurs? This is a picture of `cmake_log.txt `output by fast test.
image
And this picture is cmaked locally:
image

**Comment by abyss7** [comment]
@zlx19950903 because `hyperscan` is an optional dependency in our code, which means that you have to conditionally exclude your code if `hyperscan` is not available (i.e. checking the related macro). Your tests and code should compile and run fine in other checks.

**Comment by zlx19950903** [comment]
> @zlx19950903 because `hyperscan` is an optional dependency in our code, which means that you have to conditionally exclude your code if `hyperscan` is not available (i.e. checking the related macro). Your tests and code should compile and run fine in other checks.

If i exclude `hyperscan` from my source code, my function does not work. How about the query test? Shoule I also delete the unit test about my function?

File: src/Functions/htmlOrXmlCoarseParse.cpp [hunk 1]
```diff
@@ -0,0 +1,598 @@
...
+
+    static FunctionPtr create(const Context &) {return std::make_shared<FunctionHtmlOrXmlCoarseParse>(); }
+
+    String getName() const override {return name;}
+
+    size_t getNumberOfArguments() const override {return 1;}
+
+    DataTypePtr getReturnTypeImpl(const DataTypes & arguments) const override
+    {
+        if(!isString(arguments[0]))
```
Comment (ucasFL) [inline src/Functions/htmlOrXmlCoarseParse.cpp:567]: Style error, need a space between `if` and `(`.

**Comment by abyss7** [comment]
> > @zlx19950903 because `hyperscan` is an optional dependency in our code, which means that you have to conditionally exclude your code if `hyperscan` is not available (i.e. checking the related macro). Your tests and code should compile and run fine in other checks.
> 
> If i exclude `hyperscan` from my source code, my function does not work. How about the query test? Shoule I also delete the unit test about my function?

It's expected to not work. There is an array of skipped tests inside `docker/test/fasttest/run.sh` - add your test here.

**Comment by alexey-milovidov** [comment]
I believe this function can run faster.

This is on 80 vCPU machine:

```
:) SELECT count() FROM minicrawl WHERE NOT ignore(htmlOrXmlCoarseParse(content))

SELECT count()
FROM minicrawl
WHERE NOT ignore(htmlOrXmlCoarseParse(content))

→ Progress: 1.58 million rows, 143.44 GB (19.86 thousand rows/s., 1.80 GB/s.)
```

```
  18.86%  clickhouse              [.] runSheng
  14.56%  clickhouse              [.] roseRunProgram
  14.15%  clickhouse              [.] fdr_exec_teddy_msks4_pck
  11.27%  clickhouse              [.] roseCatchUpNfas
   6.28%  clickhouse              [.] handleSomInternal
   3.87%  clickhouse              [.] roseCatchUpAll
   3.75%  clickhouse              [.] nfaQueueExecToMatch
   3.70%  clickhouse              [.] roseFloatingCallback
   3.32%  clickhouse              [.] std::__1::vector<DB::(anonymous namespace)::HxCoarseParseImpl::SpanInfo, std::__1::allocator<DB::(anonymous namespace)::HxCoarseParseImpl::SpanInfo> >::push_back
   2.99%  clickhouse              [.] ZSTD_decompressSequences_bmi2
   2.77%  clickhouse              [.] DB::(anonymous namespace)::HxCoarseParseImpl::spanCollect
   1.15%  clickhouse              [.] roseNfaBlastAdaptor
   1.00%  clickhouse              [.] memcpy
   0.97%  clickhouse              [.] run_accel
   0.88%  clickhouse              [.] roseRunProgram_l
   0.85%  clickhouse              [.] roseNfaAdaptor
   0.83%  clickhouse              [.] nfaQueueExec
   0.78%  clickhouse              [.] shuftiExec
   0.73%  clickhouse              [.] handleSomExternal
   0.55%  clickhouse              [.] DB::deserializeBinarySSE2<4>
   0.51%  clickhouse              [.] DB::(anonymous namespace)::FunctionHtmlOrXmlCoarseParse::executeImpl
   0.45%  clickhouse              [.] nfaExecSheng_Q2
```

For `max_threads = 1`, the speed is just about 50 MB/sec. Looks obnoxiously slow.

**Pull request merged**